nexus/src/sagas.rs

```diff
@@ -852,32 +901,37 @@ fn saga_disk_delete() -> SagaTemplate<SagaDiskDelete> {
         new_action_noop_undo(sdd_delete_region_records),
     );
 
+    template_builder.append(
+        "no_result",
+        "DeleteVolumeRecord",
+        new_action_noop_undo(sdd_delete_volume_record),
+    );
```

Comment:
Deleting a disk currently immediately deletes the volume.

If start using the volume ID as a shared reference, we'll need to update this strategy (either by refcounting, or some other deferred deletion technique).